tests/Unit/Action/CancelActionTest.php

```diff
@@ -69,6 +71,38 @@ public function orderHasFinalStatusTest(): void
         $action->execute($request);
     }
 
+    /**
+     * @test
+     */
+    public function emptyOrderIdInDetails(): void
+    {
+        $action = $this->getCancelAction(
+            OrderCanceledResponse::fromResponse(
+                FileTestUtil::decodeJsonFromFile(
+                    __DIR__ . '/../../Integration/Request/data/orderCanceledResponse.json'
+                )
+            ),
+            OrderRetrieveResponse::fromResponse(
+                FileTestUtil::decodeJsonFromFile(
+                    __DIR__ . '/../../Integration/Request/data/retrieveOrderResponse.json'
+                )
+            )
+        );
+
+        $details = FileTestUtil::decodeJsonFromFile(__DIR__ . '/../../Integration/Action/data/detailsWithOrderId.json');
+        $details['orderId'] = null;
+
+        $payment = new Payment();
+        $payment->setConfigKey('pos2');
+        $payment->setDetails($details);
+        $payment->setOrderId('123');
+
+        $request = new Cancel($payment);
+        $request->setModel($payment->getDetails());
+
+        $action->execute($request);
+    }
```

Comment:
I can't see test with empty orderId in details and firstModel - we expect exception then. It's also nice to test this behavior

Reply:
done